dot/core/mocks_generate.go

```diff
@@ -0,0 +1,6 @@
+// Copyright 2022 ChainSafe Systems (ON)
```

Comment:
Please keep the *_test.go mock files like this. You can generate duplicate mocks where needed from locally defined interfaces. Otherwise this creates more package dependencies which makes cycle dependency hell a closer reality (and hard to break up packages as well).

Reply:
> Otherwise this creates more package dependencies which makes cycle dependency hell a closer reality (and hard to break up packages as well).

This particular instance does not create an cycle dependency. 

I was trying to check if there is any convention to have `_test.go` in mock files. Add some reference for such mocking guidelines in go.

I checked in other popular go projects to see if they are adding `_test.go`, 
Check the kubernetes project https://github.com/kubernetes/kubernetes/search?q=Code+generated+by+MockGen
In most cases files containing mocks don't have `_test,go` in them.
Same in some aws code https://github.com/search?l=Go&q=org%3Aaws+Code+generated+by+MockGen.&type=Code
Similar things in grafana too https://github.com/grafana/grafana/blob/a14621fff679add99c494628f1fb9a001a734f47/pkg/services/live/runstream/mock.go

Reply:
From what I understand anything that ends with `_test.go` has to be a test.

Reply:
I really don't enjoy these debates. I have added _test back, and have added a separate mock for network in babe.